### PR TITLE
feat: support load balancing of multiple addr

### PR DESCRIPTION
**Pull request by BugenZhao**
In some cases, we may want to test whether concurrent sessions from different stateless database frontends are working correctly. This PR allows users to specify multiple (host, port)s and make `sqllogictest` choose one randomly per session.

Related: https://github.com/singularity-data/risingwave/issues/4720

### CODE REVIEW

**Comment by BugenZhao** [comment]
Currently, we reuse a single session in serial mode, so only one address will be used in the whole lifetime of a `sqllogictest` execution. How about using different sessions per file, or keeping this and only making it work under parallel mode?

cc @xxchan @skyzh

**Review by xxchan — COMMENTED** [review]
IMO parallel mode tests concurrent sessions, and multiple hosts actually mean multiple db instances, which seems strange for `slt-bin`...

The concept of multiple frontend instances sharing the same backend can't be dealt with `slt-bin`. Maybe using scripts or custom binary is more appropriate 🙁

File: sqllogictest-bin/src/lib.rs [hunk 1]
```diff
@@ -61,11 +62,13 @@ struct Opt {
...
     /// The database server host.
+    /// If multiple addresses are specified, one will be chosen randomly per session.
     #[clap(short, long, default_value = "localhost")]
-    host: String,
+    host: Vec<String>,
     /// The database server port.
+    /// If multiple addresses are specified, one will be chosen randomly per session.
     #[clap(short, long, default_value = "5432")]
-    port: u16,
+    port: Vec<u16>,
```
Comment (wangrunji0408) [inline sqllogictest-bin/src/lib.rs:71]: It looks weird. We should merge them into one argument:
```suggestion
    /// If multiple addresses are specified, one will be chosen randomly per session.
    #[clap(short, long, default_value = "localhost:5432")]
    host: Vec<String>,
```

**Comment by BugenZhao** [comment]
> parallel mode tests concurrent sessions

I've updated the description to "testing multiple stateless database frontend". 🙁

> multiple frontend instances sharing the same backend

I think this is somehow common for RDBMSes? Traditional databases also support replicated instances or master-slave architecture for load balancing.

File: sqllogictest-bin/src/lib.rs (hunk 1 above)
Comment (BugenZhao) [reply sqllogictest-bin/src/lib.rs:71]: I've thought of this, but this change will break the command-line interfaces. Is this acceptable? 😢

**Comment by xxchan** [comment]
> I think this is somehow common for RDBMSes? Traditional databases also support replicated instances or master-slave architecture for load balancing.

Good point! It makes more sense to me 😄. However, for master-slave architecture, we can't simply choose host randomly...?

File: sqllogictest-bin/src/lib.rs (hunk 1 above)
Comment (wangrunji0408) [reply sqllogictest-bin/src/lib.rs:71]: Oops. How about introducing a new argument?
```rust
    /// Multiple addresses to database servers.
    /// If specified, one will be chosen randomly per session. --host and --port will be ignored.
    #[clap(long, default_value = "localhost:5432")]
    hosts: Vec<SocketAddr>,
```

Comment (BugenZhao) [reply sqllogictest-bin/src/lib.rs:71]: Actually the `tokio-postgres` interface is also designed to accept multiple hosts and ports separately. 😄

https://docs.rs/tokio-postgres/0.7.7/src/tokio_postgres/connect.rs.html#22-24

Comment (wangrunji0408) [reply sqllogictest-bin/src/lib.rs:71]: Is that mean this feature has already been supported by `tokio-postgres` so we can directly forward `host` and `port` to it?

Comment (BugenZhao) [reply sqllogictest-bin/src/lib.rs:71]: `tokio-postgres` will try to connect to each address in order if an error occurs, which differs from our demands. 😢

Comment (wangrunji0408) [reply sqllogictest-bin/src/lib.rs:71]: 😢. Anyway, since `tokio-postgres` is designed this way, it looks acceptable to me.

**Review by wangrunji0408 — APPROVED** [review]

**Pull request merged**